### DISCUSSION TITLE
学習用のテキストデータの記述形式について

### DISCUSSION THREAD

**Discussion by RRR-troisR** [espnet/espnet#3884] [Q&A] [answered]
自前の音声合成モデルを作成する際にdata.shとdataprep.shを他モデルのものを改変して流用しているので流用前のモデルとテキストデータの記述形式が一致するように人手で修正しているのですが、実際はどのくらいまでのテキストデータの記述形式に対応しているかお聞きしたいです。また、調べていた際にdata.shのみ用意すればいいという趣旨の記述があったのですが、dataprep.shは不要でしょうか？

**Comment by kan-bayashi** [comment, marked as answer]
こんにちは、林です。
Espnet2のレシピでは `local/data.sh`を実行したのち、学習用・検証用・評価用のKaldi形式のデータディレクトリが生成されるようになっていれば中はどんなふうになっていてもかまいません。
内部でデータディレクトリを整備してもいいですし、別のシェルスクリプトを呼んでもいいですし、Pythonでデータディレクトリを整備しても良いです。
極端なことを言えば、スクリプト化しなくても手動で3つのデータディレクトリを用意すれば、空の `local/data.sh` でもOKです。
データディレクトリの形式は以下を参照してください。
https://github.com/espnet/espnet/tree/master/egs2/TEMPLATE#about-kaldi-style-data-directory
https://github.com/espnet/data_example
https://kan-bayashi.github.io/asj-espnet2-tutorial/#%E3%83%87%E3%83%BC%E3%82%BF%E3%83%87%E3%82%A3%E3%83%AC%E3%82%AF%E3%83%88%E3%83%AA%E3%81%AE%E6%A7%8B%E9%80%A0
